src/Schema.js

```diff
@@ -169,47 +169,37 @@ function invalidClassNameMessage(className) {
   return 'Invalid classname: ' + className + ', classnames can only have alphanumeric characters and _, and must start with an alpha character ';
 }
 
-// Returns { error: "message", code: ### } if the type could not be
-// converted, otherwise returns a returns { result: "mongotype" }
-// where mongotype is suitable for inserting into mongo _SCHEMA collection
-function schemaAPITypeToMongoFieldType(type) {
-  var invalidJsonError = { error: "invalid JSON", code: Parse.Error.INVALID_JSON };
-  if (type.type == 'Pointer') {
-    if (!type.targetClass) {
-      return { error: 'type Pointer needs a class name', code: 135 };
-    } else if (typeof type.targetClass !== 'string') {
-      return invalidJsonError;
-    } else if (!classNameIsValid(type.targetClass)) {
-      return { error: invalidClassNameMessage(type.targetClass), code: Parse.Error.INVALID_CLASS_NAME };
-    } else {
-      return { result: '*' + type.targetClass };
-    }
-  }
-  if (type.type == 'Relation') {
-    if (!type.targetClass) {
-      return { error: 'type Relation needs a class name', code: 135 };
-    } else if (typeof type.targetClass !== 'string') {
-      return invalidJsonError;
-    } else if (!classNameIsValid(type.targetClass)) {
-      return { error: invalidClassNameMessage(type.targetClass), code: Parse.Error.INVALID_CLASS_NAME };
-    } else {
-      return { result: 'relation<' + type.targetClass + '>' };
-    }
-  }
-  if (typeof type.type !== 'string') {
-    return { error: "invalid JSON", code: Parse.Error.INVALID_JSON };
-  }
-  switch (type.type) {
-    default:         return { error: 'invalid field type: ' + type.type, code: Parse.Error.INCORRECT_TYPE };
-    case 'Number':   return { result: 'number' };
-    case 'String':   return { result: 'string' };
-    case 'Boolean':  return { result: 'boolean' };
-    case 'Date':     return { result: 'date' };
-    case 'Object':   return { result: 'object' };
-    case 'Array':    return { result: 'array' };
-    case 'GeoPoint': return { result: 'geopoint' };
-    case 'File':     return { result: 'file' };
-  }
+const invalidJsonError = new Parse.Error(Parse.Error.INVALID_JSON, "invalid JSON");
+const validNonRelationOrPointerTypes = [
+  'Number',
+  'String',
+  'Boolean',
+  'Date',
+  'Object',
+  'Array',
+  'GeoPoint',
+  'File',
+];
+// Returns an error suitable for throwing if the type is invalid
+const fieldTypeIsInvalid = ({ type, targetClass }) => {
+  if (['Pointer', 'Relation'].includes(type)) {
+    if (!targetClass) {
+      return new Parse.Error(135, `type ${type} needs a class name`);
```

Comment:
The error codes come from the Parse JS SDK which doesn't include error 135 (invalid type) :(